common/client/models.go

```diff
@@ -74,3 +75,41 @@ func (n NodeTier) String() string {
 		return fmt.Sprintf("NodeTier(%d)", n)
 	}
 }
+
+type syncIssue int
+
+func (s syncIssue) String() string {
+	if s == 0 {
+		return "synced"
+	}
+	var result bytes.Buffer
+	for i := syncIssueNotInSyncWithPool; i < syncIssueLen; i = i << 1 {
+		if i&s == 0 {
+			continue
+		}
+		result.WriteString(i.string())
+		result.WriteString(",")
+	}
+	result.Truncate(result.Len() - 1)
+	return result.String()
+}
+
+func (s syncIssue) string() string {
+	switch s {
+	case syncIssueNotInSyncWithPool:
+		return "NotInSyncWithRPCPool"
+	case syncIssueHeadIsNotIncreasing:
```

Comment:
NIT: Instead of "HeadIsNotIncreasing" and "FinalizedHeadIsNotIncreasing" we could use "NoNewHead" and "NoNewFinalizedHead". I think "HeadIsNotIncreasing" could be confused with recieving a head but the block height has already been seen